cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/FilterUil.scala

```diff
@@ -134,7 +134,9 @@ class FilterUtil(filters: Map[String, Array[Filter]]){
           false
         }else
         {
-          if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.intValue(), filters)
+          if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.longValue(), filters)
+          else if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.intValue(), filters)
+          else if (field.isInstanceOf[JsBoolean])  satisfiesAll(field.as[JsBoolean].value, filters)
           else if (field.isInstanceOf[JsBoolean])  satisfiesAll(field.as[JsBoolean].value, filters)
```

Comment:
The same is true about line 140, the control flow will never go to this line, so this line ins redundant. Not sure, why we need at JsBoolean clauses any way